File: LeetCode/python3/565.py
class Solution:
    def arrayNesting(self, nums: List[int]) -> int:
        longest = 1
        for i in range(len(nums)):
            s = set()
            count = 0
            j = i
            while nums[j] >= 0 and nums[j] != j:
                s.add(nums[j])
                nums[j] *= -1
                j = -nums[j]
                count += 1
            longest = max(longest, count)
        return longest

File: LeetCode/cpp/565.cpp
class Solution {
public:
    int arrayNesting(vector<int> &nums) {
        int maxLen = 0, i = 0, numMinus = 0, size = nums.size();
        for(int i = 0 ; i < size;++i) {
            if(nums[i] != -1) {
                int len = 0, j = i;
                while (nums[j] != -1) {
                    int temp = nums[j];
                    nums[j] = -1;
                    j = temp;
                    ++len;
                    ++numMinus;
                }
                maxLen = max(maxLen, len);
            }
        }
        return maxLen;
    }
};